### PR TITLE
[MIG] partner_vat_unique: Migration to 10.0

### PR DESCRIPTION
**Pull request by cubells**
- Add condition to avoid constraint in tests.

cc @Tecnativa

### CODE REVIEW

**Review by chienandalu — APPROVED** [review]

**Review by luismontalba — APPROVED** [review]

File: partner_vat_unique/README.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit https://odoo-community.org.
```
Comment (sergio-teruel) [inline partner_vat_unique/README.rst:59]: ¿Contributors like as module template? 
https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst#contributors

**Comment by rafaelbn** [comment]
Tested in runbot

- Go to Contact
- Create a company with name "Pepe"
- Save, you cannot, message: "The VAT False already exists in another partner."



I expected that I can create companies with empy vat as described in readme: https://github.com/OCA/partner-contact/pull/580/files#diff-cb6451433dd757e73141908890b19330R11

**Review by rafaelbn — CHANGES_REQUESTED** [review]
Please review my comment

File: partner_vat_unique/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    @api.constrains('vat')
+    def _check_vat_unique(self):
+        for record in self:
+            if record.parent_id:
+                continue
+            test_condition = (config['test_enable'] and
+                              not self.env.context.get('test_vat'))
+            if test_condition:
+                continue
+            results = self.env['res.partner'].search([
```
Comment (carlosdauden) [inline partner_vat_unique/models/res_partner.py]: Use search_count for performance.

https://www.odoo.com/documentation/10.0/reference/orm.html#odoo.models.Model.search_count

**Review by carlosdauden — CHANGES_REQUESTED** [review]

**Comment by cubells** [comment]
All is done. @rafaelbn can you retest?

**Review by carlosdauden — APPROVED** [review]

File: partner_vat_unique/README.rst (hunk 1 above)
Comment (cubells) [reply partner_vat_unique/README.rst:59]: I added contributors section @sergio-teruel . Thanks.

**Comment by rafaelbn** [comment]
@carlosdauden @cubells 

What about performance?

> uses a python constraint instead of an SQL constraint, so I will probably not use it for perf reasons

**Review by rafaelbn — APPROVED** [review]
Tested functionally :+1:

**Pull request merged**